corehq/apps/users/models.py

```diff
@@ -1678,6 +1678,16 @@ def unset_location(self):
         self.get_domain_membership(self.domain).location_id = None
         self.save()
 
+    def assign_to_locations(self, location_ids):
+        self.assigned_location_ids = location_ids
+        self.get_domain_membership(self.domain).assigned_location_ids = location_ids
+        self.save()
+
+    def unassign_all_locations(self):
+        self.assigned_location_ids = []
+        self.get_domain_membership(self.domain).assigned_location_ids = []
+        self.save()
+
```

Comment:
You could also have different methods on each class, but have them accept the same parameters, to avoid having to do `if user.is_commcare_user()` each time in the code where the updates are performed (although I'm not actually sure how often that happens).  IDK, your call.